src/components/core/GeoChart.tsx

```diff
@@ -0,0 +1,239 @@
+}
+
+export type IGeoChartInnerProps = IGeoChartProps &
+    ILoadingInjectedProps &
+    IDataSourceProviderInjectedProps &
+    WrappedComponentProps;
+
+/**
+ * Geo Chart react component
+ */
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartState> {
+    public static defaultProps: Partial<IGeoChartInnerProps> = {
+        ...commonDefaultProps,
+        onDataTooLarge: noop,
+        config: {},
+    };
+
+    private map: mapboxgl.Map;
+    private mapContainerRef: HTMLDivElement;
+    private mapDataSource: IGeoDataSourceFeatures;
+    private colorMax: number;
+    private colorMin: number;
+    private sizeMax: number;
+    private sizeMin: number;
+
+    constructor(props: IGeoChartInnerProps) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            lat: DEFAULT_LOCATION.LAT,
+            lng: DEFAULT_LOCATION.LNG,
+            zoom: DEFAULT_LOCATION.ZOOM,
+        };
+    }
+
+    public componentDidMount = () => {
+        this.loadMap();
+        this.createDataSource();
+    };
+
+    public renderVisualization = () => {
+        const style: React.CSSProperties = {};
+        return (
+            <div className="gd-geo-component" style={style}>
+                <div className="mapContainer" ref={this.setMapContainerRef} />]{this.renderLoadingOverlay()}
+            </div>
+        );
+    };
+
+    private renderLoadingOverlay = (): React.ReactNode => {
+        if (this.state.isLoading) {
+            return <LoadingComponent />;
+        }
+        return null;
+    };
+
+    //
+    // getters / setters / manipulators
+    //
+
+    private setMapContainerRef = (container: HTMLDivElement): void => {
+        this.mapContainerRef = container;
+    };
+
+    private loadMap = (): void => {
+        const { lat, lng, zoom } = this.state;
+        const map = new mapboxgl.Map({
+            container: this.mapContainerRef,
+            style: MAPBOX_STYLE,
+            center: [lng, lat],
+            zoom,
+        });
+        map.on("move", this.mapMoved);
+        this.map = map;
+    };
+
+    //
+    // working with data source
+    //
+
+    private createDataSource = (): void => {
+        createDataSource(this.props.resultSpec, this.props.getPage).then(this.dataSourceCreated);
+    };
+
+    private dataSourceCreated = (dataSource: IGeoDataSource): void => {
+        const { features, colorMax, colorMin, sizeMax, sizeMin } = dataSource;
+        if (!isEqual(features, this.mapDataSource)) {
+            this.mapDataSource = features;
+            this.colorMax = colorMax;
+            this.colorMin = colorMin;
+            this.sizeMax = sizeMax;
+            this.sizeMin = sizeMin;
+
+            const DATA_SOURCE_NAME = "countries";
+            const MAP_LAYER_NAME = "mycircle";
+
+            this.setDataSource(DATA_SOURCE_NAME, features);
+
+            this.map.addLayer({
+                id: MAP_LAYER_NAME,
+                type: "circle",
```

Comment:
`DATA_SOURCE_NAME, MAP_LAYER_NAME, type: "circle",` should be constants outside

Reply:
should archive this file and implement it fully in SD-759